hibernate-core/src/main/java/org/hibernate/internal/util/collections/CollectionHelper.java

```diff
@@ -361,4 +362,27 @@ public static Properties asProperties(Map<?,?> map) {
 		}
 	}
 
+	@SuppressWarnings( "unchecked" )
+	public static <K,V> void collectMapEntries(BiConsumer<K, V> mapEntryConsumer, Object[] mappings) {
+		// even numbered
+		assert mappings.length %2 == 0;
+
+		for ( int i = 0; i < mappings.length; i += 2 ) {
+			mapEntryConsumer.accept( (K) mappings[i], (V) mappings[i+1] );
+		}
+	}
+
+	@SuppressWarnings( "unchecked" )
+	public static <K,S> Map<K, S> asMap(Object[] elements) {
```

Comment:
It would be great to provide method accepting Collections in CollectionHelper's method (in favour of array counterpart).